apps/05Mastermind.js

```diff
@@ -8,68 +8,108 @@ prompt.start();
 var board = [];
 var solution = '';
 var letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'];
-
+var hint = [];
 function printBoard() {
-    for (var i = 0; i < board.length; i++) {
-        console.log(board[i])
-    }
+  for (var i = 0; i < board.length; i++) {
+    console.log(board[i])
+  }
 }
 
 function generateSolution() {
-    for (var i = 0; i < 4; i++) {
-        var randomIndex = getRandomInt(0, letters.length);
-        solution += letters[randomIndex];
-    }
+  for (var i = 0; i < 4; i++) {
+    var randomIndex = getRandomInt(0, letters.length);
+    solution += letters[randomIndex];
+  }
+
+  return solution;
 }
 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-    // your code here
+function generateHint(solution, guess) {
+  // your code here
+    var solutionArray = solution.split("");
+    var guessArray = guess.split("");
+    var correctLetterLocations = 0;
+  // determine correct letter location
+    for(var i=0; i<=3; i++){
```

Comment:
Usually we'd try to use a variable here like `solutionArray.length - 1` instead of `3`